node/src/components/consensus/era_supervisor.rs

```diff
@@ -1042,7 +1046,10 @@ where
                     .effect_builder
                     .announce_finalized_block(finalized_block.clone())
                     .ignore();
-                self.era_supervisor.next_block_height = finalized_block.height() + 1;
+                self.era_supervisor.next_block_height = self
+                    .era_supervisor
+                    .next_block_height
+                    .max(finalized_block.height() + 1);
```

Comment:
If the `max` version was to return a different result from the old one, then it would have meant that we finalize blocks out of order or re-finalize previously finalized blocks. This also sounds like a bug to me.

Reply:
It can happen if we join in the middle of an era: `next_block_height` gets initialized based on the latest linear chain block, but then we synchronize the protocol state from the beginning of that era, and will see all its blocks finalized again.

I was actually considering initializing it with 0 instead, but I'm not sure that's a good idea.